exo/inference/mlx/stateful_model.py

```diff
@@ -1,12 +1,12 @@
-from typing import Dict, Tuple
+from typing import Dict, Tuple, Optional
 from collections import OrderedDict
 
 import mlx.core as mx
 import mlx.nn as nn
 from mlx_lm.models.cache import make_prompt_cache
+import numpy as np
```

Comment:
I don't see lumpy used in the new code below

Reply:
TBH this was an intermediate change that I think got rolled into a cleanup commit. This whole file might be unnecessary as I hoisted the whole caching mechanism to the inference engine instead of using a subclass